code/controller/controller.ino

```diff
@@ -62,80 +64,71 @@ void setup() {
 }
 
 void loop() {
-  bool is_pressed = false;
+  // bool is_pressed = false;  // unused variable?
+  static unsigned long now=0,next_update=0;
+  now=micros();
+  if (next_update < now+now+REFRESH_INTERVAL) { // time to poll our buttons again!
+	next_update=now+REFRESH_INTERVAL;  // no matter how long it takes us to do all the crap here, our next poll will be ASAP after the refresh time expires.
```

Comment:
What is "now+now+REFRESH_INTERVAL"? Why do we need to add it twice?

Also not sure how this is supposed to work. Looks like next_update is always going to be less than now+now+REFRESH_INTERVAL?

Reply:
typo.  i hit "paste" in the wrong spot.  It should be
if (next_update < now) {
next_update=now+REFRESH_INTERVAL;

i'm not compiling this, so i have nothing to go by on whether or not it actually works.